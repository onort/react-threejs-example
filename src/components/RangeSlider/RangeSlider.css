.container {
  width: 100%;
}

.slider {
  -webkit-appearance: none;
  width: 100%;
  height: 5px;
  border-radius: 5px;
  background: var(--normal-border);
  outline: none;
  transition: all 0.4s ease-in-out;
  opacity: 0.5;
}

.slider:hover {
  opacity: 1;
}

.slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  appearance: none;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  border: 2px solid var(--primary-color);
  background: #ffffff;
  cursor: pointer;
}

.slider::-moz-range-thumb {
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background: var(--primary-color);
  cursor: pointer;
}
