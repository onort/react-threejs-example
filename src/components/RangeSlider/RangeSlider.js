import React from "react"
import PropTypes from "prop-types"

import styles from "./RangeSlider.css"

const RangeSlider = props => {
  const { currentValue, onDrag, rangeMax, rangeMin } = props
  const rangeValue = currentValue !== null ? currentValue : 0
  return (
    <div
      className={`${styles.container} ${
        props.className ? props.className : ""
      }`}
    >
      <input
        type="range"
        min={rangeMin}
        max={rangeMax}
        value={rangeValue}
        className={styles.slider}
        onChange={onDrag}
      />
    </div>
  )
}

RangeSlider.propTypes = {
  className: PropTypes.string,
  currentValue: PropTypes.number,
  onDrag: PropTypes.func.isRequired,
  rangeMax: PropTypes.number.isRequired,
  rangeMin: PropTypes.number.isRequired
}

RangeSlider.defaultProps = {
  currentValue: 0,
  rangeMin: 0
}

export default RangeSlider
