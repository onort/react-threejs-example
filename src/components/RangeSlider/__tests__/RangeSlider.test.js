import React from "react"
import { shallow } from "enzyme"

import RangeSlider from "../"

describe("<RangeSlider />", () => {
  it("should match snapshot", () => {
    const fn = jest.fn()
    const wrapper = shallow(
      <RangeSlider onDrag={fn} rangeMax={10} rangeMin={0} />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should fire onDrag function when a change occurs", () => {
    const onDrag = jest.fn()
    const wrapper = shallow(
      <RangeSlider onDrag={onDrag} rangeMax={10} rangeMin={0} />
    )
    wrapper.find("input").simulate("change")
    expect(onDrag).toBeCalledTimes(1)
  })

  it("should be able to have a custom className", () => {
    const testClassName = "testClassName"
    const fn = jest.fn()
    const wrapper = shallow(
      <RangeSlider
        className={testClassName}
        onDrag={fn}
        rangeMax={10}
        rangeMin={0}
      />
    )
    expect(wrapper.find(`.${testClassName}`).exists()).toBe(true)
  })
})
