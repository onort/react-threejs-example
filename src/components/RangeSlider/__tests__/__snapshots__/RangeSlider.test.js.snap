// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeSlider /> should match snapshot 1`] = `
<div
  className="container "
>
  <input
    className="slider"
    max={10}
    min={0}
    onChange={[MockFunction]}
    type="range"
    value={0}
  />
</div>
`;

exports[`<RangeSlider /> should match snapshot 2`] = `"<div class=\\"container \\"><input type=\\"range\\" min=\\"0\\" max=\\"10\\" value=\\"0\\" class=\\"slider\\"/></div>"`;
